crates/telio-firewall/src/firewall.rs

```diff
@@ -567,7 +561,13 @@ impl StatefullFirewall {
                 telio_log_trace!("Connection {:?} closing", key);
                 if let Some(connection) = tcp_cache.get_mut(&key) {
                     connection.rx_alive = false;
-                    if !connection.tx_alive {
+                }
+            } else if flags & TcpFlags::ACK == TcpFlags::ACK {
+                if let Some(connection) = tcp_cache.get_mut(&key) {
```

Comment:
we already have a `connection_info` so there should be no need for another `get`.

Reply:
changed